frontend/pages/[lng]/email-templates/[id].tsx

```diff
@@ -0,0 +1,255 @@
+import React, { useState, useContext } from "react"
+import { isSignedIn, isAdmin } from "/lib/authentication"
+import redirect from "/lib/redirect"
+import AdminError from "/components/Dashboard/AdminError"
+import { NextPageContext as NextContext } from "next"
+import { WideContainer } from "/components/Container"
+import { useQuery, ApolloConsumer } from "@apollo/react-hooks"
+import { SubtitleNoBackground } from "/components/Text/headers"
+import { useQueryParameter } from "/util/useQueryParameter"
+import { EmailTemplate } from "/static/types/generated/EmailTemplate"
+import { EmailTemplateQuery } from "/graphql/queries/email-templates"
+import { Paper, TextField, Button } from "@material-ui/core"
+import {
+  UpdateEmailTemplateMutation,
+  DeleteEmailTemplateMutation,
+} from "/graphql/mutations/email-templates"
+import { UpdateEmailTemplate } from "/static/types/generated/UpdateEmailTemplate"
+import { DeleteEmailTemplate } from "static/types/generated/DeleteEmailTemplate"
+import CustomSnackbar from "/components/CustomSnackbar"
+import LanguageContext from "/contexes/LanguageContext"
+import Router from "next/router"
+
+interface EmailTemplateProps {
+  admin: boolean
+}
+const EmailTemplateView = (props: EmailTemplateProps) => {
+  const { admin } = props
+  const [emailTemplate, setEmailTemplate] = useState()
+  const [name, setName] = useState()
+  const [txtBody, setTxtBody] = useState()
+  const [htmlBody, setHtmlBody] = useState()
+  const [title, setTitle] = useState()
+  const [didInit, setDidInit] = useState(false)
+
+  const id = useQueryParameter("id")
+
+  if (!admin) {
+    return <AdminError />
+  }
+
+  interface SnackbarData {
+    variant: "error" | "success" | "warning" | "error"
+    message: string
+  }
+  const [snackbarData, setSnackbarData]: [SnackbarData, any] = useState({
+    variant: "error",
+    message: "Error: Could not save",
+  })
+
+  const [isSnackbarOpen, setIsSnackbarOpen] = useState(false)
+
+  const { data, loading, error } = useQuery<EmailTemplate>(EmailTemplateQuery, {
+    variables: { id: id },
+  })
+
+  const { language } = useContext(LanguageContext)
+
+  //TODO add circular progress
+  if (loading) {
```

Comment:
Complete the TODOs before production (loading indicator and error message). There is modifiable error message component for easy / nice looking error messages.